src/sagemaker/estimator.py

```diff
@@ -569,6 +569,10 @@ def start_new(cls, estimator, inputs):
         train_args['tags'] = estimator.tags
         train_args['metric_definitions'] = estimator.metric_definitions
 
+        if isinstance(inputs, s3_input):
+            if 'InputMode' in inputs.config:
+                train_args['input_mode'] = inputs.config['InputMode']
```

Comment:
I'd add a logging statement here since it's possible that someone might explicitly define different input types (one in the estimator and one in the input) and we're ignoring one of those.

Reply:
I guess we will inevitably have to give priority to the s3_input input_mode. 
Or is there a way to identify whether the estimator's input_mode has been explicitly specified or is taking the default value?

Reply:
not at this point in the code. I agree with giving priority to the s3_input's input mode - I think adding a debug-level statement would be helpful just in case it causes any confusion down the line